vaadin-list-box-flow-parent/vaadin-list-box-flow/src/main/java/com/vaadin/flow/component/listbox/ListBoxBase.java

```diff
@@ -99,6 +100,7 @@
     @Deprecated
     public void setDataProvider(DataProvider<ITEM, ?> dataProvider) {
         this.dataProvider.set( Objects.requireNonNull(dataProvider) );
+        DataViewUtils.removeComponentFilterAndSortComparator(this);
```

Comment:
IMO this should be something that the unit test validates - resetting data makes data view filtering & sorting go away.
But let me guess, there is none to begin with for listbox ?

Reply:
And the same for each component

Reply:
This case is validated with `AbstractListDataViewListenerTest::setItems_setNewItemsToComponent_filteringAndSortingRemoved()`, which is in Flow, but should be applied to any component which list data view test inherits from `AbstractListDataViewListenerTest`

Reply:
`AbstractListDataViewListenerTest` name and the test itself were initially about listeners validation, and this remove method validation has nothing to do with listeners, but I would probably keep it here to avoid boilerplate code in component tests.